R/query-string.R

```diff
@@ -24,11 +23,11 @@ parseQS <- function(qs){
   kv <- kv[sapply(kv, length) == 2] # Ignore incompletes
 
   keys <- sapply(kv, "[[", 1)
-  keys <- unname(sapply(keys, URLdecode))
+  keys <- unname(sapply(keys, decodeURI))
```

Comment:
`decodeURI` is vectorized  so the `sapply` shouldn't be necessary.

Also, I'd suggest replacing the `sapply`s that are nearby with `lapply` and `vapply` as appropriate, to make the code more robust and easier to reason about.